Sources/ActorSingletonPlugin/ActorSingleton.swift

```diff
@@ -134,3 +52,57 @@ public enum AllocationStrategySettings {
         }
     }
 }
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: ActorSingletonRemoteCallInterceptor
+
+struct ActorSingletonRemoteCallInterceptor<Singleton: DistributedActor>: RemoteCallInterceptor where Singleton.ActorSystem == ClusterSystem {
+    let system: ClusterSystem
+    let proxy: ActorSingletonProxy<Singleton>
+
+    func interceptRemoteCall<Act, Err, Res>(
+        on actor: Act,
+        target: RemoteCallTarget,
+        invocation: inout ClusterSystem.InvocationEncoder,
+        throwing: Err.Type,
+        returning: Res.Type
+    ) async throws -> Res
+        where Act: DistributedActor,
+        Act.ID == ActorID,
+        Err: Error,
+        Res: Codable
+    {
+        // FIXME: better error handling
+        guard actor is Singleton else {
+            fatalError("Wrong interceptor")
+        }
+
+        // FIXME: can't capture inout param
+        let invocation = invocation
+        return try await self.proxy.whenLocal { __secretlyKnownToBeLocal in // TODO(distributed): this is annoying, we must track "known to be local" in typesystem instead
+            try await __secretlyKnownToBeLocal.forwardOrStashRemoteCall(target: target, invocation: invocation, throwing: throwing, returning: returning)
+        }! // FIXME: !-use
```

Comment:
> I don't know why the result is optional.

because at compile time we don't know that that actor is actually definitely local, so the signature of `whenLocal` is `-> T?`. 

So since the typesystem never knows and cannot remember "aha, this was `let x: local SomeActor`" we repeatedly and painfully keep reimplementing the whenLocal all the time when we do "framework-y stuff"... I expected this will be a pain and have been working on the `local` keyword but we don't have it yet 😢 

> I feel like I just switch, semi-automatically, to whenLocal whenever I see certain type of errors. Probably not good?

Kind of expected in "frameworky" code to be honest... That's why I kept mentioning in language work that we definitely want the `local` keyword, but it is not obvious why we need it so hard until one sees implementations like these where we're constantly forced to `whenLocal` all the time... 

In other words: I don't think it's your fault; it's expected and this is good signal and "proof" that we really need this local keyword as was expected tbh. This is good / useful, as basis / example why it matters so much.

I will review in depth tho to see if we're using it too much or not tho! I don't think it should be showing up much in "normal" user code, but in our frameworky code where we manage actors for users this is bound to show up a lot.